snapcraft/models/project.py

```diff
@@ -414,17 +492,51 @@ class App(models.CraftBaseModel):
             "never",
         ]
         | None
-    ) = None
-    install_mode: Literal["enable", "disable"] | None = None
-    slots: UniqueList[str] | None = None
-    plugs: UniqueList[str] | None = None
-    aliases: UniqueList[str] | None = None
-    environment: dict[str, str] | None = None
-    command_chain: list[str] = []
-    sockets: dict[str, Socket] | None = None
-    daemon_scope: Literal["system", "user"] | None = None
-    activates_on: UniqueList[str] | None = None
-    passthrough: dict[str, Any] | None = None
+    ) = pydantic.Field(
+        default=None, description="The condition under which the service restarts."
+    )
+    install_mode: Literal["enable", "disable"] | None = pydantic.Field(
+        default=None,
+        description="Whether snapd can automatically start the service when the snap is installed.",
+    )
+    slots: UniqueList[str] | None = pydantic.Field(
+        default=None, description="The app's slots."
+    )
+    plugs: UniqueList[str] | None = pydantic.Field(
+        default=None, description="The interfaces that the app can connect to."
+    )
+    aliases: UniqueList[str] | None = pydantic.Field(
+        default=None, description="The app's alternative internal identifiers."
+    )
+    environment: dict[str, str] | None = pydantic.Field(
+        default=None,
+        description="The runtime environment variables available to the snap's apps.",
+        examples=[
+            "environment: {PYTHONPATH: $SNAP/usr/lib/python3/dist-packages, DISABLE_WAYLAND: 1"
+        ],
+    )
+    command_chain: list[str] = pydantic.Field(
+        default_factory=list,
+        description="The sequence of commands to run before the app's command runs. Also applied when running ``snap run --shell``",
+    )
+    sockets: dict[str, Socket] | None = pydantic.Field(
+        default=None,
+        description="The app's sockets.",
+    )
+    daemon_scope: Literal["system", "user"] | None = pydantic.Field(
+        default=None,
+        description="Determines whether the service is run on a system or user instance of systemd.",
+        examples=["daemon-scope: user"],
+    )
+    activates_on: UniqueList[str] | None = pydantic.Field(
+        default=None,
+        description="The slots exposed by the snap to activate the service with D-Bus.",
+        examples=["activates-on: gnome-shell-dbus"],
+    )
+    passthrough: dict[str, Any] | None = pydantic.Field(
+        default=None,
+        description="Attributes to not validate for correctness. Useful for testing experimental snapd features.",
```

Comment:
This reads like I can list other keys in my snapcraft.yaml to avoid them from being validated.

Reply:
I agree that we need to be careful with how "attributes" could be interpreted. I'm not even sure "attribute" is correct, since we're discussing another YAML file.

What about something like:

> Values to push to the built snap's metadata file, `snap.yaml`. Snapcraft doesn't validate the values provided here, so this key is a convenient means of configuring snap features that are experimental or in early access.